packages/frontend/src/Happy/Frontend/AttrGrammar/Mangler.lhs

```diff
@@ -53,20 +57,26 @@ manipulation and let binding goop
 >                  return (rulesStr,nub (allSubProductions++prods))
 
 
->    where partitionRules a b c [] = (a,b,c)
->          partitionRules a b c (RightmostAssign attr toks : xs) = partitionRules a (SubAssign (arity,attr) toks : b) c xs
->          partitionRules a b c (x@(SelfAssign _ _ )  : xs) = partitionRules (x:a) b c xs
->          partitionRules a b c (x@(SubAssign _ _)    : xs) = partitionRules a (x:b) c xs
->          partitionRules a b c (x@(Conditional _)    : xs) = partitionRules a b (x:c) xs
+>    where arity = length lhs
+
+>          partitionRules a b c [] = (a,b,c)
+>          partitionRules a b c (RightmostAssign attr toks : xs) = partitionRules
```

Comment:
perhaps a better formatting choice would be

```
... = partitionRules a (x : b) c xs
  where x = MkAgSubAssign (arity,attr) toks
```

to align this `partitionRules` clause with the others.